### PR TITLE
Ex39/add data fetching logic to product card

### PR DESCRIPTION
**Pull request by fbozhkov**
Combined EX[38,39,40,46]

1. EX38 Add data fetching logic to SpecialCard
2. EX39 Add data fetching logic to ProductCard
3. EX40 Add data fetching logic to SearchBar
4. EX46 Create mock data sql/sequelize script (types/brands/products) with help from @petya0111 

- edits to README.md explaining how to load the mock data 
- added i18next to package.json

### CODE REVIEW

File: client/src/components/main-layout/mainPage.js [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
-    <SpecialComponent></SpecialComponent>,];
-
-    let addedTodayProducts = [<ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,];
-
+    const [products, getProducts] = useState('');
```
Comment (dplamenov) [inline client/src/components/main-layout/mainPage.js]: rename `getProducts` to `setProducts`

File: client/src/components/main-layout/mainPage.js [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
+    const categoryURL = url + '/api/types';
+
+    useEffect(() => {
+        getAllProducts();
+    }, []);
+
+    const getAllProducts = () => {
+        axios.get(productURL)
+        .then ((response) => {
+            getProducts(response.data);
```
Comment (dplamenov) [inline client/src/components/main-layout/mainPage.js]: rename `getProducts` to `setProducts`

File: client/src/components/main-layout/mainPage.js [hunk 3]
```diff
@@ -1,31 +1,46 @@
...
 import SearchBar from "../search-bar/search-bar";
-import {useTranslation} from "react-i18next";
 import ProductCard from '../product-card/ProductCard';
+import ProductList from '../product-list/ProductList';
+import SpecialProductList from '../special-product-list/SpecialProductList';
 import SpecialComponent from '../special-components/SpecialComponent';
 import './mainPage.scss';
-
+import axios from 'axios';
+import {useTranslation} from "react-i18next";
```
Comment (dplamenov) [inline client/src/components/main-layout/mainPage.js]: import axios and react-i18next before local file imports

File: client/src/components/main-layout/mainPage.js (hunk 1 above)
Comment (dplamenov) [inline client/src/components/main-layout/mainPage.js]: default value must be empty array []

File: client/src/components/main-layout/mainPage.js [hunk 4]
```diff
@@ -1,31 +1,46 @@
...
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,
-    <ProductCard></ProductCard>,];
-
+    const [products, getProducts] = useState('');
+
+    const url = 'http://localhost:3004'; // change url when deploying
+    const productURL = url + '/api/products';
+    const categoryURL = url + '/api/types';
```
Comment (dplamenov) [inline client/src/components/main-layout/mainPage.js]: use string template literals

File: client/src/components/product-list/ProductList.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
```
Comment (dplamenov) [inline client/src/components/product-list/ProductList.js:1]: same issues from ProductCars.js

File: client/src/components/product-list/ProductList.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import ProductCard from '../product-card/ProductCard';
+import Grid from '@mui/material/Grid';
+import axios from 'axios';
+import '../main-layout/mainPage.scss';
+import {useTranslation} from "react-i18next";
+
+const ProductList = () => {
+    let {t} = useTranslation()
```
Comment (dplamenov) [inline client/src/components/product-list/ProductList.js]: make this const

File: client/src/components/product-list/ProductList.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            const products = response.data;
+            getProducts(products);
+        })
+        .catch(error => console.error(`Error: ${error}`));
+    }
+
+    const productKeys = Object.values(products);
+    const addedTodayProducts = productKeys.map( productData => 
+         
+        <ProductCard {...productData} /> );
```
Comment (dplamenov) [inline client/src/components/product-list/ProductList.js]: fix code format

File: client/src/components/search-bar/search-bar.js [hunk 8]
```diff
@@ -7,15 +7,33 @@ import Select from '@mui/material/Select';
...
+    const getAllCategories = () => {
+        axios.get(categoryURL)
+        .then ((response) => {
+            const responseCategories = response.data;
+            getCategories(responseCategories);
+        })
+        .catch(error => console.error(`Error: ${error}`));
+    }
+    const arr = [];
+    Object.keys(categories).forEach(key => arr.push({name: key, value: categories[key]}));
```
Comment (dplamenov) [inline client/src/components/search-bar/search-bar.js]: use map instead forEach: 

`
const arr = Object.keys(categories).map(key => ({name: key, value: categories[key]}));
`

**Review by dplamenov — CHANGES_REQUESTED** [review]

**Comment by fbozhkov** [comment]
Changes made @dplamenov

**Review by petya0111 — APPROVED** [review]
Great work for building it up altogether! ✔

**Review by JordanRad — APPROVED** [review]

**Review by JordanRad — APPROVED** [review]

**Pull request merged**